src/main/java/org/redcrosswarriors/controllerservice/RequestBloodControllerService.java

```diff
@@ -0,0 +1,257 @@
+import org.redcrosswarriors.repository.RequestBloodDetailsRepository;
+import org.redcrosswarriors.repository.RequestTimeDetailsRepository;
+import org.redcrosswarriors.repository.UserDetailsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class RequestBloodControllerService
+{
+    @Autowired
+    private RequestBloodDetailsRepository requestRepository;
+
+    @Autowired
+    private RequestTimeDetailsRepository requestTimeRepository;
+
+    private ReturnForRequestBlood returnRequest = new ReturnForRequestBlood();
+
+    private RequestedTimeDetails timeDetails;
+
+    @Transactional
+    public boolean requestBlood(RequestBloodInput requestInput)
+    {
+        boolean check = true;
+        try{
+            System.out.println("start");
+            requestRepository.newRequester(requestInput.getFirstName(),
+                    requestInput.getLastName(),
+                    requestInput.getEmail(),
+                    requestInput.getPhoneNumber(),
+                    requestInput.getBloodType(),
+                    requestInput.getHospitalName(),
+                    requestInput.getStreetName(),
+                    requestInput.getCityName(),
+                    requestInput.getStateName(),
+                    requestInput.getZipCode(),
+                    requestInput.getMessage());
+
+
+            System.out.println(requestInput.getLastName());
+            System.out.println(requestInput.getEmail());
+            System.out.println(requestInput.getPhoneNumber());
+            System.out.println(requestInput.getBloodType());
+            System.out.println(requestInput.getHospitalName());
+            System.out.println(requestInput.getStreetName());
+            System.out.println(requestInput.getCityName());
+            System.out.println(requestInput.getStateName());
+            System.out.println( requestInput.getZipCode());
+            System.out.println( requestInput.getMessage());
+
+
+
+            timeDetails = requestTimeRepository.getRequestedTimeByEmail(requestInput.getEmail());
+            String currentTime;
+            if(timeDetails == null)
+            {
+                currentTime = returnTime();
+                System.out.println(currentTime);
+                requestTimeRepository.requesterTimeUpdate(requestInput.getEmail(), currentTime);
+                List<String> matches;
+
+                matches = requestRepository.findMatches(requestInput.getBloodType());
+                if(matches.size() == 0)
+                {
+                    System.out.println("0 matches near by");
+                    //check = false;
+
+                }
```

Comment:
This if statement seems pointless since it's just printing that there are 0 matches.

Reply:
okay will do that